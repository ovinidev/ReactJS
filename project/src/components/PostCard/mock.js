export const PostCardPropsMock = {
  title: "algum title",
  body: "algum body",
  id: 1,
  cover: "img/img.png"
};