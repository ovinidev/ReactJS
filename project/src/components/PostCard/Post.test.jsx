import { render, screen } from "@testing-library/react";
import { PostCard } from ".";
import { PostCardPropsMock } from "./mock";

const props = PostCardPropsMock;

describe("<PostCard />", () => {
  it("should render PostCard correctly", () => {
    render(<PostCard {...props} />);

    expect(screen.getByRole("img", { name: "algum title" }))
    .toHaveAttribute("src", props.cover);

    expect(screen.getByRole("heading", { name: "algum title 1"})).toBeInTheDocument();

    expect(screen.getByText("algum body")).toBeInTheDocument();

  });

  it("should match snapshot", () => {
    const { container } = render(<PostCard {...props} />);

    expect(container.firstChild).toMatchSnapshot();

  });
});