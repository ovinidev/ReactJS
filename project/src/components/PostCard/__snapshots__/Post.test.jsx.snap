// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<PostCard /> should match snapshot 1`] = `
<div
  class="post"
>
  <img
    alt="algum title"
    src="img/img.png"
  />
  <div
    class="posts-content"
  >
    <h1>
      algum title
       
      1
    </h1>
    <p>
      algum body
    </p>
    <p>
      u
    </p>
  </div>
</div>
`;
